app/controllers/sitemap_controller.rb

```diff
@@ -0,0 +1,47 @@
+class SitemapController < ApplicationController
+
+  def sitemap
+    #if false
+      #redirect_to "/" # The CDN then fetches the sitemap.xml from the sitemap/generate URL which points to the `generate` action
+    #else
+    render_site_map
+    #end
+  end
+
+  def generate
+    render_site_map
+  end
+
+  private
+
+    def render_site_map
+      respond_to do |format|
+
+        data = Rails.cache.fetch(@current_community, expires_in: 24.hours) do
+          adapter = SitemapGenerator::NeverWriteAdapter.new
+
+          if @current_community.use_domain 
+            default_host = "https://"+@current_community.domain
+          else 
+            default_host = "https://"+APP_CONFIG.domain
+          end
+
+          community_id = @current_community.id
+
+          SitemapGenerator::Sitemap.create(
+                :default_host => default_host,
+                :verbose => false,
+                :adapter => adapter) do
+                  Listing.where(deleted: false, open: true, community_id: community_id).find_each do |listing|
```

Comment:
Couple more ideas:

We've been doing some decoupling of controller code and ActiveModels. You've probably seen those "Service" layer implementations here and there in the code. In this case, could you add a new method to controller, say, `find_open_listings(community_id)` which would return an array of listing IDs and update timestamps? Then the controller code would be at least a bit more decoupled from ActiveModel, because the controller code doesn't care anymore whether the data is coming from database, cache, file, over the network e.g. as long as it returns an array of IDs and timestamps.

* Should we add some upper limit to the listings? Like, 5000 newest open listings? If there's a marketplace which has 10000000 open listings and we would fetch them all and render the sitemap, that would be terrible for performance.